js/views/inbox_view.js

```diff
@@ -6,34 +6,78 @@
 
     window.Whisper = window.Whisper || {};
 
-    var SocketView = Whisper.View.extend({
-        className: 'status',
+    Whisper.NetworkStatusView = Whisper.View.extend({
+        className: 'network-status',
         initialize: function() {
-            setInterval(this.updateStatus.bind(this), 5000);
-        },
-        updateStatus: function() {
-            var className, message = '';
-            if (typeof getSocketStatus === 'function') {
-              switch(getSocketStatus()) {
-                  case WebSocket.CONNECTING:
-                      className = 'connecting';
-                      break;
-                  case WebSocket.OPEN:
-                      className = 'open';
-                      break;
-                  case WebSocket.CLOSING:
-                      className = 'closing';
-                      break;
-                  case WebSocket.CLOSED:
-                      className = 'closed';
-                      message = i18n('disconnected');
-                      break;
-              }
-            if (!this.$el.hasClass(className)) {
-                this.$el.attr('class', className);
-                this.$el.text(message);
+            this.$el.hide();
+
+            setInterval(this.render.bind(this), 5000);
+            setTimeout(this.finishConnectingGracePeriod.bind(this), 5000);
```

Comment:
Is this grace period really necessary? This means that if we're still 'CONNECTING' 5 seconds after the window is first opened, we'll see the banner before/until the connection eventually completes or fails?